packages/adblocker/src/engine/engine.ts

```diff
@@ -877,6 +880,8 @@ export default class FilterEngine extends EventEmitter<EngineEventHandlers> {
       allowSpecificHides = shouldApplyHideException(specificHides) === false;
     }
 
+    domain ||= '';
```

Comment:
it is used the same way above https://github.com/ghostery/adblocker/pull/3881/commits/8c268f746a4c59d6185323cb9f50fedd00ac7c5b

how about we move default value to the argument directly ?

Reply:
From `tldts` lib, the result of `parsed` domain name can be `null`. If we don't want to refine the value at this level, we should change the type completely.

```ts
import { IOptions, IResult } from 'tldts-core';
export declare function parse(url: string, options?: Partial<IOptions>): IResult;
```
```ts
export interface IResult {
    hostname: string | null;
    isIp: boolean | null;
    subdomain: string | null;
    domain: string | null;
    publicSuffix: string | null;
    domainWithoutSuffix: string | null;
    isIcann: boolean | null;
    isPrivate: boolean | null;
}
```